src/Core/Picking.js

```diff
@@ -187,9 +188,16 @@ export default {
                 // if baseId matches objId, the clicked point belongs to `o`
                 for (let i = 0; i < candidates.length; i++) {
                     if (candidates[i].objId == o.baseId) {
+                        // Compute distance to the camera
+                        const pointCoordinate = new Coordinates(view.referenceCrs, o.position.x, o.position.y, o.position.z);
```

Comment:
I think this method for computing distance could cause issues : the `position` property for threejs `Object3D` is expressed in the object's parent coordinates system.
Here, the point could be centered within a `Group`, and the `Group` could be positioned at the correct world coordinates. the `position` of the point would therefore be `(0, 0, 0)`. Using world position for the point, as you did for the camera would prevent this.

Reply:
Also, do you know of an usage of this picking method other than with `PointCloudLayer` ?

Reply:
Yes, you're right, I added the world transformation and updated a bit this part to fix a bug that I had not seen for point cloud picking.